lxd/main_init_interactive.go

```diff
@@ -360,16 +360,27 @@ func (c *cmdInit) askNetworking(config *api.InitPreseed, d lxd.InstanceServer) e
 					continue
 				}
 
-				// Add to the default profile
-				config.Node.Profiles[0].Devices["eth0"] = map[string]string{
-					"type":    "nic",
-					"nictype": "macvlan",
-					"name":    "eth0",
-					"parent":  interfaceName,
-				}
+				// Add to the default profile.
+				network, _, err := d.GetNetwork(interfaceName)
```

Comment:
we should probably differentiate a not found error from other errors here and only continue on a not found error.

We shouldnt mask other errors and treat them as not found.